docs/en/stack/ml/setup.asciidoc

```diff
@@ -5,100 +5,140 @@
-roles instead and set your {kib} privileges appropriately for each space.
+Each {ml} job can be assigned to all, one or multiple spaces. This can be 
+configured in **{stack-manage-app}** > _Alerts and Insights_ > 
+_Machine Learning Jobs_. To assign a job to a space, select the spaces icon 
+shown in the job list.
 
-For full access to the {ml-features} in {kib}, you must have:
+[role="screenshot"]
+image::assign-job-spaces.jpg["Assign machine learning jobs to spaces"]
+
+
+[discrete]
+[[kib-security-privileges]]
+=== {kib} feature privileges
+
+Within a {kib} space, for full access to the {ml-features}, you must have:
 
 [%interactive]
-* [ ] `all` {kib} privileges for the {ml-features} in the appropriate spaces
-* [ ] `read` and `view_index_metadata` index privileges on source indices
-* [ ] `read`, `manage`, and `index` index privileges on destination indices (for
-  {dfanalytics-jobs} only)
+* [ ] `All` {kib} privileges for the {ml-features}
+* [ ] {data-sources} for your source indices
+* [ ] `read`, and `view_index_metadata` index privileges on your source indices
+* [ ] {data-sources}, `read`, `manage`, and `index` index privileges on 
+  destination indices (for {dfanalytics-jobs} only)
 
-For read-only access to the {ml-features} in {kib}, you must have:
+
+Within a {kib} space, for read-only access to the {ml-features}, you must have:
 
 [%interactive]
-* [ ] `read` {kib} privileges for the {ml-features} in the appropriate spaces
-* [ ] `read` index privileges on source indices
-* [ ] `read` index privileges on destination indices (for {dfanalytics-jobs}
-only)
+* [ ] `Read` {kib} privileges for the {ml-features}
+* [ ] {data-sources} for your source indices
+* [ ] `read` index privilege on your source indices
+* [ ] {data-sources} and `read` index privilegeson destination indices (for 
+  {dfanalytics-jobs} only)
+
+NOTE: {data-sources-cap} can be automatically created when creating a 
+{dfanalytics-job}.
+
+For access to use {ml} APIs via _Dev Tools_ in {kib}, set the {es} security 
+privileges and grant access to `machine_learning_admin` or 
+`machine_learning_user` built-in roles.
 
+
+[discrete]
 [[upload-file-security-privileges]]
-To upload and import files in the *{data-viz}*, you must have:
+=== Security privileges for {data-viz}
+
+Within a {kib} space, to upload and import files in the *{data-viz}*, you must 
+have:
 
 [%interactive]
-* [ ] `all` {kib} privileges for the {ml-app} or Discover features in
-the appropriate spaces. Alternatively, `read` {kib} privileges for the {ml-app}
-feature and `all` {kib} privileges for the {ipm-app} feature
-* [ ] `manage_pipeline` or `manage_ingest_pipelines` cluster privileges
-* [ ] `create`, `create_index`, `manage` and `read` index privileges for
-destination indices
+- [ ] Kibana feature privileges that grant use of File Upload, either:
+  * `All` Machine Learning or
+  * `All` Discover or
+  * `Read` Machine Learning and `All` Data View Management 
```

Comment:
I've tested all of the combinations and couldn't get it working with just `All` for Discover (and ofc all required index privileges). Not sure where this requirement came from, but we should double check what to expect - maybe we have a bug here.

Reply:
Thank you for your help @pheyos! I'd like to backport these changes to 7.16. Just to double-check: does the security information apply to 7.16 as well?

Reply:
@szabosteve I think we have some small differences in 7.16, but let me double check.

Reply:
The changes we had from 7.16 to 8.0 are only around ML now being part of Kibana's `base` privileges. We tell the reader that `All Kibana privileges for the machine learning features` or `Read Kibana privileges for the machine learning features` are required, but we don't mention HOW to set these, so our documentation is valid for both versions and I can't spot anything that would need a 7.16 specific change.